massgov/mayflower
# DP-11360 a11y: Audit Listing Table & Make Improvements

Any PRs being created needs a changelog.txt file before being merged into dev. See: [Change Log Instructions](https://github.com/massgov/mayflower/blob/dev/docs/change-log-instructions.md)


## Description

### React
- Fix the `aria-expanded` state update on an accordion button.
- Fix the button icon style to match Patternlab (when an accordion is open the chevron icon on its corresponding button points upward.)
- Remove the hardcoded class "collapsed" from `div.ma__listing-table__extra` which is supposed to be dynamically managed which causes screenreaders to min-announce the button state.
- Change the markup from table to list to improve its semantic structure and resolve the Voiceover reading issue.
- Add a heading for the listing table.

### Patternlab
- Change the markup from table to list to improve its semantic structure and resolve the Voiceover reading issue.
- Add a heading for the listing table.

## Related Issue / Ticket

- [DP-11360 a11y: Audit Listing Table & Make Improvements](https://jira.mass.gov/browse/DP-11360)

## Steps to Test


1. Go to molecule/listing table in Storybook.

2. Click the accordion buttons to see:
     1. their chevron icon on the right change where it points as you click:
          - when accordion is close, the icon points DOWNWARD.
          - when accordion is open, the icon points UPWARD.
     2. their `aria-expanded` value reflects their corresponding accordion's state.
          - open the dev tool, and find the button in the source. Find `aria-expanded` in the button element.
          - see the `aria-expanded` value changes as you click the button.
               - when accordion is close, the value is FALSE.
               - when accordion is open, the value is TRUE.
          - listen how the button is announced in Voiceover as you click it. when you locates the button with [control + option + right arrow key], [control + option + space] to click the button. (Voiceover gives you the instruction to do so when you get to the button). 
               - when accordion is close, it announces "show more COLLAPSED button" .
               - when accordion is open, it announces  "show less EXPENDED button".

3. In source, check `div.ma__listing-table__extra` doesn't have an extra "collapsed" in its class.


## Screenshots
**Accordion button icon**
State: accordion is CLOSE.
<img width="415" alt="listing_table-accordion" src="https://user-images.githubusercontent.com/9633303/49593631-044b4680-f942-11e8-87a0-26e2bc7374a2.png">

State: accordion is OPEN.
BEFORE
<img width="386" alt="listing_table-before" src="https://user-images.githubusercontent.com/9633303/49593670-1d53f780-f942-11e8-8f65-1da6e2876d26.png">

AFTER
<img width="461" alt="listing_table-after" src="https://user-images.githubusercontent.com/9633303/49593680-2349d880-f942-11e8-9603-c4e7b679a789.png">


## Additional Notes:

Anything else to add?

#### Impacted Areas in Application


* 

#### @TODO


*

#### Today I learned...
